pkg/controller/llmisvc/router_discovery_filter.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package llmisvc
+
+import (
+	"net"
+	"strings"
+
+	"knative.dev/pkg/apis"
+)
+
+type URLPredicateFn func(*apis.URL) bool
+
+func Filter(urls []*apis.URL, predicate URLPredicateFn) []*apis.URL {
+	if len(urls) == 0 {
+		return []*apis.URL{}
+	}
+
+	result := make([]*apis.URL, 0, len(urls))
+	for _, url := range urls {
+		if predicate(url) {
+			result = append(result, url)
+		}
+	}
+	return result
+}
+
+func IsInternalURL(url *apis.URL) bool {
+	host := url.Host
+
+	if colonIndex := strings.LastIndex(host, ":"); colonIndex != -1 {
+		host = host[:colonIndex]
+	}
+
+	if isInternalIP(host) {
+		return true
+	}
+
+	return isInternalHostname(host)
+}
+
+func IsExternalURL(url *apis.URL) bool {
+	return !IsInternalURL(url)
+}
+
+func FilterInternalURLs(urls []*apis.URL) []*apis.URL {
+	return Filter(urls, IsInternalURL)
+}
+
+func FilterExternalURLs(urls []*apis.URL) []*apis.URL {
+	return Filter(urls, IsExternalURL)
+}
+
+func isInternalIP(addr string) bool {
+	ip := net.ParseIP(addr)
+	if ip != nil && ip.IsPrivate() {
+		return true
+	}
+	return false
+}
+
+func isInternalHostname(hostname string) bool {
+	hostname = strings.ToLower(hostname)
+
+	return hostname == "localhost" || strings.HasSuffix(hostname, ".local") ||
```

Comment:
instead of `.local.` I think we might need to use `network.GetClusterDomainName()` that handles the case of a different kube internal local domain

Reply:
I saw the lookup of `"/etc/resolv.conf` while looking at existing code and I wonder at which point I forgot about changing it. Thanks for the pointer!